### DISCUSSION TITLE
How to select FI option ?

### DISCUSSION THREAD

**Discussion by IdrissSinapan** [obss/sahi#867] [Q&A] [answered]
Hello,
In the paper we have for example, TOOD + SF + SAHI + FI + PO or TOOD + SF + SAHI + FI etc. 
How can I choose the option I want when I am using SAHI please ? I would like to use YoloV7 + SF + SAHI + FI + PO. 
Thank you !

**Comment by anuragrawal2024** [comment]
@IdrissSinapan Did you figure this out? I want to do YOLOv8 + SAHI + FI + PO for one of my projects. Is there an option to do SAHI + FI together or will I have to make some code changes? My understanding is that I can do SAHI and FI separately and concatenate their results before postprocessing/NMS but before I make code changes, I wanted to check if this has already been implemented. Thank you for your help!  

@fcakyon Can you please advise?

**Comment by anuragrawal2024** [comment, marked as answer]
Never mind! I found this in the get_sliced_prediction():

perform_standard_pred: bool
Perform a standard prediction on top of sliced predictions to increase large object
detection accuracy. Default: True.

**Reply by IdrissSinapan** [reply]
Hi @utility-aagrawal, no I did not figure out.. I want to use YOLOV7 + SAHI + FI but I think that I am using only YOLOV7 + SAHI. You are saying that if I use the get_sliced_prediction() and do not perform a standard prediction (by setting "False") it will use FI ? 
In the official paper it is written "Optionally, predictions from full inference can also be added." So if it is optional, this means that we have to change the default value... no ?

**Comment by anuragrawal2024** [comment]
@IdrissSinapan , If you are using get_sliced_prediction(), you'll need to set perform_standard_pred to True and which is its default value. If you look at the below piece of code from predict.py, this parameter enables FI after performing SAHI:

![image](https://github.com/obss/sahi/assets/140737044/2854c7d8-de0d-4281-be86-dcf927d64edd)

**Reply by IdrissSinapan** [reply]
Hi @utility-aagrawal, thank you very much for your answer, I will try this.
